### PR TITLE
LSTM and fix a bug

### PR DESCRIPTION
**Pull request by Gloria98**

### CODE REVIEW

File: rl755/models/common/lstm.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        self.hidden_size = hidden_size
+        self.lstm = tf.keras.Sequential([
+            tf.keras.layers.Dense(units=self.hidden_size, activation=None),
+            tf.keras.layers.LSTM(units=self.hidden_size, return_sequences=True)
+        ])
+        self.final_layer = tf.keras.layers.Dense(
+            units=self.output_size, activation=None
+        )
+
+    def call(self, inputs, training=None):
```
Comment (mmatena) [inline rl755/models/common/lstm.py]: I think you'll need to add a `mask=None` parameter here. Due to the autoregressive nature of the LSTM, you're probably fine with doing nothing with it.

File: rl755/models/common/lstm.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        x = self.lstm(inputs, training=training)
+        self.hidden_output = x
+        output = self.final_layer(x, training=training)
+        return output
+
+    def get_loss_fn(self):
+        """Train using a MSE loss."""
+        return tf.keras.losses.MeanSquaredError()
+
+    def get_hidden_representation(self, x, training=None):
```
Comment (mmatena) [inline rl755/models/common/lstm.py]: The interface for this has changed a bit. It's now:

```python
def get_hidden_representation(self, x, mask=None, training=None, position=-1):
        raise NotImplementedError()
```
I think you just have to change this to:

```python
def get_hidden_representation(self, x, mask=None, training=None, position=-1):
    self(x, training=training, mask=mask)
    return self.hidden_output[..., position, :]
```

File: rl755/models/common/lstm.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import tensorflow as tf
+from rl755.models.common.sequential import SequentialModel
+
+class LSTM(SequentialModel):
+    def __init__(self, hidden_size, output_size, **kwargs):
+        super().__init__(**kwargs)
+        self.output_size = output_size
+        self.hidden_size = hidden_size
+        self.lstm = tf.keras.Sequential([
+            tf.keras.layers.Dense(units=self.hidden_size, activation=None),
```
Comment (mmatena) [inline rl755/models/common/lstm.py]: Do we need the dense layer here? I'm unfamiliar with LSTMs, so feel free to leave it in if its here for a reason.

**Review by mmatena — APPROVED** [review]
Thanks for the PR! Sorry for my lateness in reviewing it.

Overall looks pretty good. Feel free to merge it in once you've made the changes.

File: rl755/models/common/lstm.py (hunk 3 above)
Comment (Gloria98) [reply rl755/models/common/lstm.py]: It's there not for some specific reasons, just based on some examples I saw online. I think I will delete it.
